Various strictDeps and cross fixes

Artturin opened the pull request:
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Artturin committed:
librsvg: fix binfmt cross-compiling

librsvg-aarch64-unknown-linux-gnu> error: linker `cc` not found
librsvg-aarch64-unknown-linux-gnu>   |
librsvg-aarch64-unknown-linux-gnu>   = note: No such file or directory (os error 2)
librsvg-aarch64-unknown-linux-gnu> error: could not compile `librsvg` due to previous error
librsvg-aarch64-unknown-linux-gnu> make[2]: *** [Makefile:1572: /build/librsvg-2.52.0/target/aarch64-unknown-linux-gnu/release/rsvg-convert] Error 101

Artturin committed:
help2man: fix build with strictDeps

and enable strictDeps
perlPackages.perl is needed in nativeBuildInputs because of
the configuration script

Artturin committed:
groff: fix strictDeps by moving defining perl path out of the
cross-compile conditional

Artturin committed:
apparmor: disable strictDeps explicitly for libapparmor

do not override python to python3 in all-packages

Artturin committed:
itstool: enable strictDeps

Artturin committed:
gdk-pixbuf: enable strictDeps

gdk-pixbuf: make gobject-introspection conditional to fix binfmt cross-compile

Artturin committed:
libnotify: enable strictDeps

checked output with diffoscope

libnotify: make gobject-introspection conditional to fix binfmt cross-compile

Artturin committed:
apparmor: enable strictDeps for apparmor-utils

checked with diffoscope

Artturin committed:
xcb-util-xrm: enable strictDeps

Artturin committed:
perlPackages.X11XCB: fix strictDeps

Artturin committed:
pango: disable strictDeps when building with introspection

pango: fix binfmt cross-compile

Artturin committed:
graphviz: fix build with strictDeps

Artturin committed:
opensp: enable strictDeps and remove unneeded deps

neither gentoo or arch list them
checked with diffoscope and there were no changes

Artturin committed:
docbook2x: fix building with strictDeps

output checked with diffoscope

Artturin committed:
git: fix build with strictDeps

Artturin committed:
wayland: fix build with strictDeps

Artturin committed:
python3Packages.xpybutil: disable tests

there are no tests

Artturin committed:
i3: fix build with strictDeps

Artturin committed:
libgudev: enable strictDeps

libgudev: make gobject-introspection conditional to fix binfmt cross-compile

Artturin committed:
dconf: fix build with strictDeps

Artturin committed:
autogen: fix build with strictDeps

Artturin committed:
pulseaudio: fix build with strictDeps

Artturin committed:
sway: fix binfmt cross-compilation

SuperSandro2000 reviewed (commented):
Can we finally fix gobject-introspection for cross compile instead of patching around it in so many packages?

Artturin committed:
gi-docgen: fix cross by adding python3 to depsBuildBuild

adding python3 to nativeBuildInputs is not enough as it still uses the
aarch64 python for the build

Artturin commented:
should be ready to merge

Artturin committed:
pax-utils: enable strictDeps

Mindavi commented:
Please mark as ready to review. I'll (try to) check in a few hours what the status of this is on my hydra, since I saw some things appeared broken yesterday.

jonringer reviewed (approved):
diff lgtm

cc @jtojnar for gobject-introspection changes

Artturin commented:
moved scdoc to depsBuildBuild instead of having it in both nativeBuildInputs and buildInputs

Mindavi commented:
Presumably I just hit a flaky build for libuv, it seems to build ok now. Queue now needs to clear, but in general this is probably at least not a regression :).

jtojnar commented on pkgs/development/libraries/gdk-pixbuf/default.nix:
```diff
@@ -58,7 +62,18 @@ stdenv.mkDerivation rec {
+
+  buildInputs = [
+    python3
+    gi-docgen
```
Why is this in `buildInputs`? Same with g-i.

Artturin replied:
found out that only pkg-config has to be in depsBuildBuild and "it just works"

jtojnar commented on pkgs/development/tools/documentation/gi-docgen/default.nix:
```diff
@@ -29,6 +29,12 @@ python3.pkgs.buildPythonApplication rec {
     })
   ];
 
+  strictDeps = true;
```
Python packages are strict by default

jtojnar commented on pkgs/development/libraries/gdk-pixbuf/default.nix:
```diff
@@ -43,13 +44,17 @@ stdenv.mkDerivation rec {
+  # gdk-pixbuf-thumbnailer is not wrapped therefore strictDeps will work
+  strictDeps = true;
+
+  depsBuildBuild = [ pkg-config ];
```
Could you please keep the dep-per-line in files that use it?

jtojnar commented on pkgs/development/libraries/libnotify/default.nix:
```diff
@@ -27,15 +28,24 @@ stdenv.mkDerivation rec {
+    gobject-introspection
+  ];
+
+  buildInputs = lib.optionals withIntrospection [
```
Why in `buildInputs`?

Artturin replied:
if i apply
 
```diff
diff --git a/pkgs/development/libraries/libnotify/default.nix b/pkgs/development/libraries/libnotify/default.nix
index 40089e3d7e9..cf5dd487e75 100644
--- a/pkgs/development/libraries/libnotify/default.nix
+++ b/pkgs/development/libraries/libnotify/default.nix
@@ -33,6 +33,10 @@ stdenv.mkDerivation rec {
 
   strictDeps = true;
 
+  depsBuildBuild = [
+    pkg-config
+  ];
+
   nativeBuildInputs = [
     meson
     ninja
@@ -44,10 +48,6 @@ stdenv.mkDerivation rec {
     gobject-introspection
   ];
 
-  buildInputs = lib.optionals withIntrospection [
-    gobject-introspection
-  ];
-
   propagatedBuildInputs = [
     gdk-pixbuf
     glib
```

then the build fails with 
```
libnotify> [14/15] Generating Notify-0.7.gir with a custom command0.p/libnotify.so.4.0.0.symbols
libnotify> FAILED: libnotify/Notify-0.7.gir
libnotify> /nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/bin/g-ir-scanner --no-libtool --namespace=Notify --nsversion=0.7 --warn-all --output libnotify/Notify-0.7.gir --c-include=libnotify/notify.h -I/build/libnotify-0.7.9/libnotify -I/build/libnotify-0.7.9/build/libnotify -I./. -I../. --filelist=/build/libnotify-0.7.9/build/libnotify/libnotify.so.4.0.0.p/Notify_0.7_gir_filelist --include=GLib-2.0 --include=GdkPixbuf-2.0 --pkg-export=libnotify --cflags-begin -I./. -I../. '-DG_LOG_DOMAIN="libnotify"' -I/nix/store/wbvvi6jsrpm368ga6x7wd2j7z3b6yrhs-glib-2.70.0-dev/include -I/nix/store/wbvvi6jsrpm368ga6x7wd2j7z3b6yrhs-glib-2.70.0-dev/include/glib-2.0 -I/nix/store/hglhdivhj6dya46xx05bsmr68azwp92d-glib-2.70.0/lib/glib-2.0/include -I/nix/store/1ji5kw304sdwvdb9asqniwmrdmkx3rxg-gdk-pixbuf-2.42.6-dev/include/gdk-pixbuf-2.0 -I/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/include/gobject-introspection-1.0 --cflags-end --add-include-path=/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0 -L./libnotify --library notify -L/nix/store/hglhdivhj6dya46xx05bsmr68azwp92d-glib-2.70.0/lib -L/nix/store/xk2is79zbvmslhb1bw4dk12j1jbyy2l6-gdk-pixbuf-2.42.6/lib --extra-library=gdk_pixbuf-2.0 --extra-library=gobject-2.0 --extra-library=glib-2.0 --extra-library=gio-2.0 -L/nix/store/rd8na9jgqgvp4k1razyxlxdi15416l57-gobject-introspection-1.70.0/lib --extra-library=girepository-1.0 --sources-top-dirs /build/libnotify-0.7.9/subprojects/ --sources-top-dirs /build/libnotify-0.7.9/build/subprojects/
libnotify> Couldn't find include 'GdkPixbuf-2.0.gir' (search path: '['/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0', '/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share', '/nix/store/dvjlviry8rrcrgaxj63q95wxzxagjbpg-pkg-config-wrapper-0.29.2/share/gir-1.0', '/nix/store/fvwk5kbk1rifxjva4j2j40isj80v5yb7-meson-0.57.1/share/gir-1.0', '/nix/store/y5v8rk457phrv71jc9slx3a58lyvdmll-ninja-1.10.2/share/gir-1.0', '/nix/store/1dr9x27m5mxqrv0rx2i9ckz9m34gkvfw-libxslt-1.1.34-dev/share/gir-1.0', '/nix/store/bzaw10zf10wf3ws5xp59zkvmp0xg0chn-libxslt-1.1.34-bin/share/gir-1.0', '/nix/store/kybxplf3r7rbbaajsr4zwwvcbj23dxf8-docbook-xsl-ns-1.79.2/share/gir-1.0', '/nix/store/wbvvi6jsrpm368ga6x7wd2j7z3b6yrhs-glib-2.70.0-dev/share/gir-1.0', '/nix/store/gxwpxk04i6appwhv5zsmx2rm4zjr10bd-zlib-1.2.11/share/gir-1.0', '/nix/store/vw7v3r8zgxp5kgc6m736zb98ybbrcfa4-gettext-0.21/share/gir-1.0', '/nix/store/hglhdivhj6dya46xx05bsmr68azwp92d-glib-2.70.0/share/gir-1.0', '/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0', '/nix/store/c7mwhb56d18aj4nl0g2hs2203cn6z5pk-patchelf-0.13/share/gir-1.0', '/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0', '/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0', '/usr/share/gir-1.0']')
libnotify> ninja: build stopped: subcommand failed.
```

jtojnar replied:
Hmm, that sounds like the `gobject-introspection` setup hook not setting `XDG_DATA_DIRS` properly.

Mindavi commented:
Sway and librsvg are fixed with this, also for the native case (since last force-push)

Mindavi commented:
I feel like the combination of strictDeps and gi are not very nice, but I'm not sure why.

Artturin commented on pkgs/development/libraries/pango/default.nix:
```diff
@@ -34,12 +34,21 @@ stdenv.mkDerivation rec {
     sha256 = "IeH1eYvN/adeq8QoBRSwiWq1b2VtTn5mAwuaJTXs3Jg=";
   };
 
+  strictDeps = !withIntrospection;
```
```
pango> [97/152] Generating Pango-1.0.gir with a custom command00.10.p/libpango-1.0.so.0.4800.10.symbols[K
pango> FAILED: pango/Pango-1.0.gir
pango> /nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/bin/g-ir-scanner --no-libtool --namespace=Pango --nsversion=1.0 --warn-all --output pango/Pango-1.0.gir --c-include=pango/pango.h --quiet -I/build/pango-1.48.10/pango -I/build/pango-1.48.10/build/pango -I./. -I../. -I./pango/. -I../pango/. --filelist=/build/pango-1.48.10/build/pango/libpango-1.0.so.0.4800.10.p/Pango_1.0_gir_filelist --include=HarfBuzz-0.0 --include=GObject-2.0 --include=cairo-1.0 --symbol-prefix=pango --identifier-prefix=Pango --pkg-export=pango --cflags-begin -D_POSIX_C_SOURCE=200809L -D_POSIX_THREAD_SAFE_FUNCTIONS -D_GNU_SOURCE -I./. -I../. -I./pango/. -I../pango/. -I/nix/store/wbvvi6jsrpm368ga6x7wd2j7z3b6yrhs-glib-2.70.0-dev/include/glib-2.0 -I/nix/store/hglhdivhj6dya46xx05bsmr68azwp92d-glib-2.70.0/lib/glib-2.0/include -I/nix/store/wbvvi6jsrpm368ga6x7wd2j7z3b6yrhs-glib-2.70.0-dev/include -I/nix/store/bdn4pwhfy0fclc7h768ni8gnhl029wmq-fribidi-1.0.10/include/fribidi -I/nix/store/iarxmv8cv59f8zmdf22ywacl56wr7syx-libthai-0.1.28-dev/include -I/nix/store/9fgrfhjrhwmaxpl3qlw29n9axmg2wb2y-harfbuzz-2.8.2-dev/include/harfbuzz -I/nix/store/hf26kkychqkwh7xij0kg59glsjshkzjz-fontconfig-2.13.94-dev/include -I/nix/store/21g6l30yh4x09hbkpk302dp85ikyrwfm-freetype-2.11.0-dev/include/freetype2 -I/nix/store/1lbxskz6sn91kh2vrrmf4qscp7zxcmmw-xorgproto-2021.5/include -I/nix/store/0hi9f9lf8xv35r5wifnqak3m0v0fk9xs-libXrender-0.9.10-dev/include -I/nix/store/vmxvjxi972fpd6bvc8dispb1i77cq3am-libX11-1.7.2-dev/include -I/nix/store/f26d1agr7zjz0x3rgc3d1fc25p7riagq-libXft-2.3.4-dev/include -I/nix/store/i89g3lb6ndbgfw3dgzdkgc7ncqp2cs18-cairo-1.16.0-dev/include/cairo -I/nix/store/21g6l30yh4x09hbkpk302dp85ikyrwfm-freetype-2.11.0-dev/include -I/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/include/gobject-introspection-1.0 --cflags-end --add-include-path=/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0 -L./pango --library pango-1.0 --extra-library=m -L/nix/store/hglhdivhj6dya46xx05bsmr68azwp92d-glib-2.70.0/lib --extra-library=glib-2.0 --extra-library=gobject-2.0 --extra-library=gio-2.0 -L/nix/store/bdn4pwhfy0fclc7h768ni8gnhl029wmq-fribidi-1.0.10/lib --extra-library=fribidi -L/nix/store/2d7qnp3whrbr41mp9mhf2ipimb04arnb-libthai-0.1.28/lib --extra-library=thai -L/nix/store/kffy24h2iphb93rrmm6kvd28bl24947f-harfbuzz-2.8.2/lib --extra-library=harfbuzz -L/nix/store/634ia5si5cr9qd1kim5jflwkvczg3v4l-fontconfig-2.13.94-lib/lib -L/nix/store/hib60qcr4dl53x1g376fpqfrc02w0jrv-freetype-2.11.0/lib --extra-library=fontconfig --extra-library=freetype -L/nix/store/z2i3lnq92k05vrjhifai1vdczxpr8wpx-libXrender-0.9.10/lib -L/nix/store/cvknal47kqp9b4lndn678wc70wiiykf2-libX11-1.7.2/lib --extra-library=Xrender --extra-library=X11 -L/nix/store/420d42rc4iyjbsg11lca110afc764vqw-libXft-2.3.4/lib --extra-library=Xft -L/nix/store/q1x1sj9kwrnjn0qk1607knxyay6mah2d-cairo-1.16.0/lib --extra-library=cairo --extra-library=harfbuzz-gobject -L/nix/store/rd8na9jgqgvp4k1razyxlxdi15416l57-gobject-introspection-1.70.0/lib --extra-library=girepository-1.0 --sources-top-dirs /build/pango-1.48.10/subprojects/ --sources-top-dirs /build/pango-1.48.10/build/subprojects/
pango> Couldn't find include 'HarfBuzz-0.0.gir' (search path: '['/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0', '/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share', '/nix/store/dvjlviry8rrcrgaxj63q95wxzxagjbpg-pkg-config-wrapper-0.29.2/share/gir-1.0', '/nix/store/fvwk5kbk1rifxjva4j2j40isj80v5yb7-meson-0.57.1/share/gir-1.0', '/nix/store/y5v8rk457phrv71jc9slx3a58lyvdmll-ninja-1.10.2/share/gir-1.0', '/nix/store/wbvvi6jsrpm368ga6x7wd2j7z3b6yrhs-glib-2.70.0-dev/share/gir-1.0', '/nix/store/gxwpxk04i6appwhv5zsmx2rm4zjr10bd-zlib-1.2.11/share/gir-1.0', '/nix/store/vw7v3r8zgxp5kgc6m736zb98ybbrcfa4-gettext-0.21/share/gir-1.0', '/nix/store/hglhdivhj6dya46xx05bsmr68azwp92d-glib-2.70.0/share/gir-1.0', '/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0', '/nix/store/rwlxhp31mfkazphzfyfzlvvkqxywqfpm-gi-docgen-2021.8/share/gir-1.0', '/nix/store/dmff68nlqclkjwk7x2ss7ayk9xgxycwl-python3-3.9.6/share/gir-1.0', '/nix/store/c7mwhb56d18aj4nl0g2hs2203cn6z5pk-patchelf-0.13/share/gir-1.0', '/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0', '/nix/store/8z27iv5k1sbm2vkrq3y50g0pbajrcrhh-gobject-introspection-1.70.0-dev/share/gir-1.0', '/usr/share/gir-1.0']')
pango> [98/152] Generating symbol file pango/libpango-1.0.so.0.4800.10.p/libpango-1.0.so.0.4800.10.symbols
pango> ninja: build stopped: subcommand failed.
```

Artturin commented:
rebased onto staging and running nixpkgs-review for sway and flashfocus

Artturin commented:
Result of `nixpkgs-review pr 142602` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>flashfocus</li>
    <li>sway</li>
  </ul>
</details>

Artturin commented:
how would i build only the changed packages without using nixpkgs-review -p

Mindavi commented:
Specify them all with `nix-build --keep-going -A pkgsCross.aarch64-multiplatform.pkg1 -A pkgsCross.aarch64-multiplatform.pkg2 -A ...`. Think that should work.

Artturin commented:
running
```
nix build .#pkgsCross.aarch64-multiplatform.git .#pkgsCross.aarch64-multiplatform.i3 .#pkgsCross.aarch64-multiplatform.sway .#pkgsCross.aarch64-multiplatform.dconf .#pkgsCross.aarch64-multiplatform.gdk-pixbuf .#pkgsCross.aarch64-multiplatform.libgudev .#pkgsCross.aarch64-multiplatform.libnotify .#pkgsCross.aarch64-multiplatform.librsvg .#pkgsCross.aarch64-multiplatform.pango .#pkgsCross.aarch64-multiplatform.wayland .#pkgsCross.aarch64-multiplatform.python3Packages.xpybutil .#pkgsCross.aarch64-multiplatform.gi-docgen .#pkgsCross.aarch64-multiplatform.autogen .#pkgsCross.aarch64-multiplatform.help2man .#pkgsCross.aarch64-multiplatform.itstool .#pkgsCross.aarch64-multiplatform.ghostscript .#pkgsCross.aarch64-multiplatform.libapparmor .#pkgsCross.aarch64-multiplatform.pax-utils .#pkgsCross.aarch64-multiplatform.pulseaudio .#pkgsCross.aarch64-multiplatform.xcbutilxrm .#pkgsCross.aarch64-multiplatform.graphviz .#pkgsCross.aarch64-multiplatform.groff .#pkgsCross.aarch64-multiplatform.opensp .#pkgsCross.aarch64-multiplatform.docbook2x .#pkgsCross.aarch64-multiplatform.perlPackages.X11XCB --keep-going
```

Artturin commented:
result of 
```
nixpkgs-review pr -p git -p i3 -p sway -p dconf -p gdk-pixbuf -p libgudev -p libnotify -p librsvg -p pango -p wayland -p python3Packages.xpybutil -p gi-docgen -p autogen -p help2man -p itstool -p ghostscript -p libapparmor -p pax-utils -p pulseaudio -p xcbutilxrm -p graphviz -p groff -p opensp -p docbook2x -p perlPackages.X11XCB
```


Result of `nixpkgs-review pr 142602` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>25 packages built:</summary>
  <ul>
    <li>autogen</li>
    <li>dconf</li>
    <li>docbook2x</li>
    <li>gdk-pixbuf</li>
    <li>ghostscript</li>
    <li>gi-docgen</li>
    <li>git</li>
    <li>graphviz</li>
    <li>groff</li>
    <li>help2man</li>
    <li>i3</li>
    <li>itstool</li>
    <li>libapparmor</li>
    <li>libgudev</li>
    <li>libnotify</li>
    <li>librsvg</li>
    <li>opensp</li>
    <li>pango</li>
    <li>pax-utils</li>
    <li>perlPackages.X11XCB</li>
    <li>pulseaudio</li>
    <li>python3Packages.xpybutil</li>
    <li>sway</li>
    <li>wayland</li>
    <li>xcbutilxrm</li>
  </ul>
</details>

Artturin commented:
```
[artturin@nixos:~/nixpkgs]$ nix build .#pkgsCross.aarch64-multiplatform.git .#pkgsCross.aarch64-multiplatform.i3 .#pkgsCross.aarch64-multiplatform.sway .#pkgsCross.aarch64-multiplatform.dconf .#pkgsCross.aarch64-m
ultiplatform.gdk-pixbuf .#pkgsCross.aarch64-multiplatform.libgudev .#pkgsCross.aarch64-multiplatform.libnotify .#pkgsCross.aarch64-multiplatform.librsvg .#pkgsCross.aarch64-multiplatform.pango .#pkgsCross.aarch64-
multiplatform.wayland .#pkgsCross.aarch64-multiplatform.python3Packages.xpybutil .#pkgsCross.aarch64-multiplatform.gi-docgen .#pkgsCross.aarch64-multiplatform.autogen .#pkgsCross.aarch64-multiplatform.help2man .#p
kgsCross.aarch64-multiplatform.itstool .#pkgsCross.aarch64-multiplatform.ghostscript .#pkgsCross.aarch64-multiplatform.libapparmor .#pkgsCross.aarch64-multiplatform.pax-utils .#pkgsCross.aarch64-multiplatform.puls
eaudio .#pkgsCross.aarch64-multiplatform.xcbutilxrm .#pkgsCross.aarch64-multiplatform.graphviz .#pkgsCross.aarch64-multiplatform.groff .#pkgsCross.aarch64-multiplatform.opensp .#pkgsCross.aarch64-multiplatform.doc
book2x .#pkgsCross.aarch64-multiplatform.perlPackages.X11XCB
[2/17/493 built, 3 copied (8.4/8.5 MiB), 1.5 MiB DL] building aarch64-unknown-linux-gnu-binutils-2.35.2 (buildP
error: interrupted by the user

[artturin@nixos:~/nixpkgs]$ nix build .#pkgsCross.aarch64-multiplatform.git .#pkgsCross.aarch64-multiplatform.i3 .#pkgsCross.aarch64-multiplatform.sway .#pkgsCross.aarch64-multiplatform.dconf .#pkgsCross.aarch64-m
ultiplatform.gdk-pixbuf .#pkgsCross.aarch64-multiplatform.libgudev .#pkgsCross.aarch64-multiplatform.libnotify .#pkgsCross.aarch64-multiplatform.librsvg .#pkgsCross.aarch64-multiplatform.pango .#pkgsCross.aarch64-
multiplatform.wayland .#pkgsCross.aarch64-multiplatform.python3Packages.xpybutil .#pkgsCross.aarch64-multiplatform.gi-docgen .#pkgsCross.aarch64-multiplatform.autogen .#pkgsCross.aarch64-multiplatform.help2man .#p
kgsCross.aarch64-multiplatform.itstool .#pkgsCross.aarch64-multiplatform.ghostscript .#pkgsCross.aarch64-multiplatform.libapparmor .#pkgsCross.aarch64-multiplatform.pax-utils .#pkgsCross.aarch64-multiplatform.puls
eaudio .#pkgsCross.aarch64-multiplatform.xcbutilxrm .#pkgsCross.aarch64-multiplatform.graphviz .#pkgsCross.aarch64-multiplatform.groff .#pkgsCross.aarch64-multiplatform.opensp .#pkgsCross.aarch64-multiplatform.doc
book2x .#pkgsCross.aarch64-multiplatform.perlPackages.X11XCB --keep-going
error: builder for '/nix/store/714cays925py5vi7zfk22xawiw2xmp8l-libxml2-aarch64-unknown-linux-gnu-2.9.12.drv' failed with exit code 1;
       last 10 log lines:
       > make[2]: Entering directory '/build/libxml2-2.9.12/fuzz'
       > make[2]: Nothing to be done for 'all'.
       > make[2]: Leaving directory '/build/libxml2-2.9.12/fuzz'
       > Making all in xstc
       > make[2]: Entering directory '/build/libxml2-2.9.12/xstc'
       > make[2]: Nothing to be done for 'all'.
       > make[2]: Leaving directory '/build/libxml2-2.9.12/xstc'
       > make[1]: Leaving directory '/build/libxml2-2.9.12'
       > installing
       > substitute(): ERROR: file 'python/libxml2mod.la' does not exist
       For full logs, run 'nix log /nix/store/714cays925py5vi7zfk22xawiw2xmp8l-libxml2-aarch64-unknown-linux-gnu-2.9.12.drv'.
error: builder for '/nix/store/ccja3m6mkfr4421kcf2k9zl30jb8hn21-perl5.34.0-ExtUtils-PkgConfig-aarch64-unknown-linux-gnu-1.16.drv' failed with exit code 2;
       last 10 log lines:
       > *** ExtUtils::PkgConfig requires the pkg-config utility, but it doesn't
       > *** seem to be in your PATH.  Is it correctly installed?
       > ***  PATH=/nix/store/x3nyj69alqqzm68a014z38xbf7g533i0-aarch64-unknown-linux-gnu-pkg-config-wrapper-0.29.2/bin:/nix/store/kck6s2ixxxz4s5265pn0n1b1mn3ni626-perl-5.34.0-aarch64-unknown-linux-gnu-mini/bin:/ni
x/store/c7mwhb56d18aj4nl0g2hs2203cn6z5pk-patchelf-0.13/bin:/nix/store/rdn4xpgg6sgzajgsyswhpr2l7l6x8xzq-aarch64-unknown-linux-gnu-stage-final-gcc-debug-wrapper-9.3.0/bin:/nix/store/nh2lxgg5ygf1hfr3sc6m5744f6f4xhg9-
aarch64-unknown-linux-gnu-stage-final-gcc-debug-9.3.0/bin:/nix/store/08h5javjz33sk2y3kjvgz25qjwijzz5k-glibc-aarch64-unknown-linux-gnu-2.33-56-bin/bin:/nix/store/lk9kdd0nnky0d46yirl1f1fv76rhlk6g-coreutils-9.0/bin:/
nix/store/kfq3mlp14xj8bmqjimsf33mdj9h0rf7p-aarch64-unknown-linux-gnu-binutils-wrapper-2.35.2/bin:/nix/store/rgp09g3i6x5c3dg8iwkrqgixxwlfl4r5-aarch64-unknown-linux-gnu-binutils-2.35.2/bin:/nix/store/lk9kdd0nnky0d46
yirl1f1fv76rhlk6g-coreutils-9.0/bin:/nix/store/ln78ynparnypvmp7sjnkh9fjbpw8y2hq-findutils-4.8.0/bin:/nix/store/3hiz94m6fwhagmf2iiblcfdarg0sjzkp-diffutils-3.8/bin:/nix/store/m0wza2i9l6ch0k2p84b916zz1ayc7rav-gnused-
4.8/bin:/nix/store/nxaf2ph82bbj0cd0jb6qaa2vdk9ag0bj-gnugrep-3.7/bin:/nix/store/c49dkb82xbclvrlw4vmfp5fq72jab5hl-gawk-5.1.0/bin:/nix/store/zacy182lns3q7shd2gq04bd5ia7dcn56-gnutar-1.34/bin:/nix/store/aw4m1yw5ib2yvl2
bq4s5lplfhz09ivh4-gzip-1.11/bin:/nix/store/qzccgvkxkc1skggkf8a79srlbsisyv5z-bzip2-1.0.6.0.2-bin/bin:/nix/store/5nk90b0ms88syb5n9irgbyhll75c11ia-gnumake-4.3/bin:/nix/store/mvybrwylvigp7ll290z5v5dc7g6fa4v8-bash-5.1-
p8/bin:/nix/store/1r9yj4m7zz5qcs7w57f8j8j3sk7bjj0c-patch-2.7.6/bin:/nix/store/qymbfpn05i7i1123ynfap2n2myp6xxwh-xz-5.2.5-bin/bin
       > ***
       > no configure script, doing nothing
       > building
       > no Makefile, doing nothing
       > installing
       > install flags: SHELL=/nix/store/mvybrwylvigp7ll290z5v5dc7g6fa4v8-bash-5.1-p8/bin/bash pkgconfigdir=/nix/store/k25p2v3y397z7pmibri2b0fgsp1gfbzl-perl5.34.0-ExtUtils-PkgConfig-aarch64-unknown-linux-gnu-1.16/
lib/pkgconfig m4datadir=/nix/store/k25p2v3y397z7pmibri2b0fgsp1gfbzl-perl5.34.0-ExtUtils-PkgConfig-aarch64-unknown-linux-gnu-1.16/share/aclocal aclocaldir=/nix/store/k25p2v3y397z7pmibri2b0fgsp1gfbzl-perl5.34.0-ExtU
tils-PkgConfig-aarch64-unknown-linux-gnu-1.16/share/aclocal install
       > make: *** No rule to make target 'install'.  Stop.
       For full logs, run 'nix log /nix/store/ccja3m6mkfr4421kcf2k9zl30jb8hn21-perl5.34.0-ExtUtils-PkgConfig-aarch64-unknown-linux-gnu-1.16.drv'.
error: builder for '/nix/store/nv4qrc0x8m6jysafdwb3ny5i83al2qw1-perl5.34.0-IO-Tty-aarch64-unknown-linux-gnu-1.15.drv' failed with exit code 255;
       last 10 log lines:
       >   dynamic loading or has the IO module statically linked into it.)
       >  at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Handle.pm line 268.
       > Compilation failed in require at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Handle.pm line 268.
       > BEGIN failed--compilation aborted at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Handle.pm line 268.
       > Compilation failed in require at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Seekable.pm line 100.
       > BEGIN failed--compilation aborted at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Seekable.pm line 100.
       > Compilation failed in require at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/File.pm line 132.
       > BEGIN failed--compilation aborted at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/File.pm line 132.
       > Compilation failed in require at Makefile.PL line 10.
       > BEGIN failed--compilation aborted at Makefile.PL line 10.
       For full logs, run 'nix log /nix/store/nv4qrc0x8m6jysafdwb3ny5i83al2qw1-perl5.34.0-IO-Tty-aarch64-unknown-linux-gnu-1.15.drv'.
error: builder for '/nix/store/304syrard3c6w98qqiy5kav5gm9sakvz-perl5.34.0-Module-Build-aarch64-unknown-linux-gnu-0.4231.drv' failed with exit code 2;
       last 10 log lines:
       > inc_bundling_support....disabled
       >   requires:
       >     ! inc::latest is not installed
       >
       > ERRORS/WARNINGS FOUND IN PREREQUISITES.  You may wish to install the versions
       > of the modules indicated above before proceeding with this installation
       >
       > Attempt to reload Scalar/Util.pm aborted.
       > Compilation failed in require at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/Data/Dumper.pm line 314.
       > Couldn't run Build.PL: No such file or directory at lib/Module/Build/Compat.pm line 341.
       For full logs, run 'nix log /nix/store/304syrard3c6w98qqiy5kav5gm9sakvz-perl5.34.0-Module-Build-aarch64-unknown-linux-gnu-0.4231.drv'.
error: builder for '/nix/store/d7dbwwpccghj0b3hsaig53yf3z36k8d4-perl5.34.0-Params-Util-aarch64-unknown-linux-gnu-1.102.drv' failed with exit code 255;
       last 10 log lines:
       > BEGIN failed--compilation aborted at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Handle.pm line 268.
       > Compilation failed in require at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Seekable.pm line 100.
       > BEGIN failed--compilation aborted at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Seekable.pm line 100.
       > Compilation failed in require at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/File.pm line 132.
       > BEGIN failed--compilation aborted at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/File.pm line 132.
       > Compilation failed in require at inc/latest/private.pm line 10.
       > BEGIN failed--compilation aborted at inc/latest/private.pm line 10.
       > Compilation failed in require at inc/latest.pm line 5.
       > Compilation failed in require at Makefile.PL line 16.
       > BEGIN failed--compilation aborted at Makefile.PL line 16.
       For full logs, run 'nix log /nix/store/d7dbwwpccghj0b3hsaig53yf3z36k8d4-perl5.34.0-Params-Util-aarch64-unknown-linux-gnu-1.102.drv'.
error: builder for '/nix/store/f1952ma0pznqr0cpij4cq2ikgx90aicc-perl5.34.0-Test-LeakTrace-aarch64-unknown-linux-gnu-0.16.drv' failed with exit code 255;
       last 10 log lines:
       > patching ./xt/04_synopsis.t...
       > Can't load module Fcntl, dynamic loading not available in this perl.
       >   (You may need to build a new perl executable which either supports
       >   dynamic loading or has the Fcntl module statically linked into it.)
       >  at inc/Module/Install/Makefile.pm line 6.
       > Compilation failed in require at inc/Module/Install/Makefile.pm line 6.
       > BEGIN failed--compilation aborted at inc/Module/Install/Makefile.pm line 6.
       > Compilation failed in require at inc/Module/Install.pm line 309.
       > The 'makemaker_args' method does not exist in the 'inc' path!
       > Please remove the 'inc' directory and run Makefile.PL again to load it.
       For full logs, run 'nix log /nix/store/f1952ma0pznqr0cpij4cq2ikgx90aicc-perl5.34.0-Test-LeakTrace-aarch64-unknown-linux-gnu-0.16.drv'.
error: 1 dependencies of derivation '/nix/store/2wvravvh731cfrff391iyi5knf081blg-perl5.34.0-Module-Runtime-aarch64-unknown-linux-gnu-0.016.drv' failed to build
error: 1 dependencies of derivation '/nix/store/23qazc8hycywcqnxw2kkfm95z44brqhd-perl5.34.0-Parse-RecDescent-aarch64-unknown-linux-gnu-1.967015.drv' failed to build
error: 1 dependencies of derivation '/nix/store/p9q8ij2r15j5l9vvr2ada0jyzwynpx0w-perl5.34.0-Tie-IxHash-aarch64-unknown-linux-gnu-1.23.drv' failed to build
error: builder for '/nix/store/apq7brk4ybsa4vlj0xxvwd1z0x0gfa0i-perl5.34.0-Test-utf8-aarch64-unknown-linux-gnu-1.02.drv' failed with exit code 255;
       last 10 log lines:
       > patching ./xt/003perlcritic.t...
       > Can't load module Fcntl, dynamic loading not available in this perl.
       >   (You may need to build a new perl executable which either supports
       >   dynamic loading or has the Fcntl module statically linked into it.)
       >  at inc/Module/Install/Makefile.pm line 6.
       > Compilation failed in require at inc/Module/Install/Makefile.pm line 6.
       > BEGIN failed--compilation aborted at inc/Module/Install/Makefile.pm line 6.
       > Compilation failed in require at inc/Module/Install.pm line 307.
       > The 'makemaker_args' method does not exist in the 'inc' path!
       > Please remove the 'inc' directory and run Makefile.PL again to load it.
       For full logs, run 'nix log /nix/store/apq7brk4ybsa4vlj0xxvwd1z0x0gfa0i-perl5.34.0-Test-utf8-aarch64-unknown-linux-gnu-1.02.drv'.
error: 1 dependencies of derivation '/nix/store/2a7byqvyalqk1yzvpbqrl02dv0hgvsvv-perl5.34.0-Data-OptList-aarch64-unknown-linux-gnu-0.110.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5j466p349h31ph15as0ibrd38jg9c5dv-perl5.34.0-Test-File-aarch64-unknown-linux-gnu-1.443.drv' failed to build
error: 1 dependencies of derivation '/nix/store/mr3fvzs76x3fxms42ydpnbrrfmi2fj6q-perl5.34.0-Sub-Exporter-aarch64-unknown-linux-gnu-0.987.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9v149aj44k29h8si1zhv160vlljck21z-perl5.34.0-Cwd-Guard-aarch64-unknown-linux-gnu-0.05.drv' failed to build
error: builder for '/nix/store/pmy6yaji46kak42zs7kq3wsxqyj0qxw3-perl5.34.0-File-ShareDir-aarch64-unknown-linux-gnu-1.118.drv' failed with exit code 255;
       last 10 log lines:
       > BEGIN failed--compilation aborted at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Handle.pm line 268.
       > Compilation failed in require at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Seekable.pm line 100.
       > BEGIN failed--compilation aborted at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Seekable.pm line 100.
       > Compilation failed in require at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/File.pm line 132.
       > BEGIN failed--compilation aborted at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/File.pm line 132.
       > Compilation failed in require at inc/latest/private.pm line 10.
       > BEGIN failed--compilation aborted at inc/latest/private.pm line 10.
       > Compilation failed in require at inc/latest.pm line 5.
       > Compilation failed in require at Makefile.PL line 18.
       > BEGIN failed--compilation aborted at Makefile.PL line 18.
       For full logs, run 'nix log /nix/store/pmy6yaji46kak42zs7kq3wsxqyj0qxw3-perl5.34.0-File-ShareDir-aarch64-unknown-linux-gnu-1.118.drv'.
error: builder for '/nix/store/764pw1ykkzdwysa3c6d3gyqwa9d6qsbm-perl5.34.0-Devel-GlobalDestruction-aarch64-unknown-linux-gnu-0.14.drv' failed with exit code 255;
       last 10 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/w8gnmyp5mrxkj10m1fmadjc1pnm0liy6-Devel-GlobalDestruction-0.14.tar.gz
       > source root is Devel-GlobalDestruction-0.14
       > setting SOURCE_DATE_EPOCH to timestamp 1477952747 of file Devel-GlobalDestruction-0.14/README
       > patching sources
       > updateAutotoolsGnuConfigScriptsPhase
       > configuring
       > Undefined subroutine &ExtUtils::HasCompiler::tempdir called at inc/ExtUtils/HasCompiler.pm line 16.
       > Compilation failed in require at Makefile.PL line 5.
       > BEGIN failed--compilation aborted at Makefile.PL line 5.
       For full logs, run 'nix log /nix/store/764pw1ykkzdwysa3c6d3gyqwa9d6qsbm-perl5.34.0-Devel-GlobalDestruction-aarch64-unknown-linux-gnu-0.14.drv'.
error: 2 dependencies of derivation '/nix/store/pivda72qf57n90zb4dcxaab8b9yn2gjl-perl5.34.0-Module-Build-Tiny-aarch64-unknown-linux-gnu-0.039.drv' failed to build
error: 2 dependencies of derivation '/nix/store/ai8jm73sikgihd6gkdrcqc2bx0jd3v8p-perl5.34.0-Devel-CheckCompiler-aarch64-unknown-linux-gnu-0.07.drv' failed to build
error: 2 dependencies of derivation '/nix/store/zq2py3rsa4h1dciwpscps5bj4x6mwhfn-perl5.34.0-Readonly-aarch64-unknown-linux-gnu-2.05.drv' failed to build
error: 2 dependencies of derivation '/nix/store/9yr15c99ilvpyrffs1yhxqvnh36l8gfl-perl5.34.0-IPC-Run-aarch64-unknown-linux-gnu-20200505.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/dw849nxlamnxb6xix9xadxd20hraapjr-perl5.34.0-B-C-aarch64-unknown-linux-gnu-1.57.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5xzqqyx3rvk4w5424pjckkw2civ26hrw-perl5.34.0-Sub-Name-aarch64-unknown-linux-gnu-0.26.drv' failed to build
error: 3 dependencies of derivation '/nix/store/r3lxkb9ifvlivpg7n23fwx3827jzpwmi-perl5.34.0-Module-Build-XSUtil-aarch64-unknown-linux-gnu-0.19.drv' failed to build
error: builder for '/nix/store/1627z6csaj2c14bkwvlkl762pwlnknmc-perl5.34.0-XML-SAX-aarch64-unknown-linux-gnu-1.02.drv' failed with exit code 2;
       last 10 log lines:
       > Installing /nix/store/9b3sc4gia3q4dvri3m6yz81rnbzqs7bk-perl5.34.0-XML-SAX-aarch64-unknown-linux-gnu-1.02/share/man/man3/XML::SAX.3
       > Installing /nix/store/9b3sc4gia3q4dvri3m6yz81rnbzqs7bk-perl5.34.0-XML-SAX-aarch64-unknown-linux-gnu-1.02/share/man/man3/XML::SAX::ParserFactory.3
       > Installing /nix/store/9b3sc4gia3q4dvri3m6yz81rnbzqs7bk-perl5.34.0-XML-SAX-aarch64-unknown-linux-gnu-1.02/share/man/man3/XML::SAX::PurePerl.3
       > Installing /nix/store/9b3sc4gia3q4dvri3m6yz81rnbzqs7bk-perl5.34.0-XML-SAX-aarch64-unknown-linux-gnu-1.02/share/man/man3/XML::SAX::Intro.3
       > Installing /nix/store/9b3sc4gia3q4dvri3m6yz81rnbzqs7bk-perl5.34.0-XML-SAX-aarch64-unknown-linux-gnu-1.02/share/man/man3/XML::SAX::PurePerl::Reader.3
       > Installing /nix/store/9b3sc4gia3q4dvri3m6yz81rnbzqs7bk-perl5.34.0-XML-SAX-aarch64-unknown-linux-gnu-1.02/share/man/man3/XML::SAX::DocumentLocator.3
       > Appending installation info to /nix/store/9b3sc4gia3q4dvri3m6yz81rnbzqs7bk-perl5.34.0-XML-SAX-aarch64-unknown-linux-gnu-1.02/lib/perl5/5.34.0/aarch64-linux/perllocal.pod
       > Can't locate XML/SAX.pm in @INC (you may need to install the XML::SAX module) (@INC contains: /nix/store/kck6s2ixxxz4s5265pn0n1b1mn3ni626-perl-5.34.0-aarch64-unknown-linux-gnu-mini/lib/perl5/cross_perl/5.
34.0 /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0 /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-li
nux /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/site_perl/5.34.0 /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/site_per
l/5.34.0/aarch64-linux /nix/store/593fixiry0r7xalij9zk99wbxd87107b-perl5.34.0-XML-NamespaceSupport-aarch64-unknown-linux-gnu-1.12/lib/perl5/site_perl/5.34.0 /nix/store/593fixiry0r7xalij9zk99wbxd87107b-perl5.34.0-X
ML-NamespaceSupport-aarch64-unknown-linux-gnu-1.12/lib/perl5/site_perl/5.34.0/aarch64-linux /nix/store/2wib6r2p6ym61hv2jvnnraj3n48vs0pv-perl5.34.0-XML-SAX-Base-aarch64-unknown-linux-gnu-1.09/lib/perl5/site_perl/5.
34.0 /nix/store/2wib6r2p6ym61hv2jvnnraj3n48vs0pv-perl5.34.0-XML-SAX-Base-aarch64-unknown-linux-gnu-1.09/lib/perl5/site_perl/5.34.0/aarch64-linux /nix/store/9b3sc4gia3q4dvri3m6yz81rnbzqs7bk-perl5.34.0-XML-SAX-aarch
64-unknown-linux-gnu-1.02/lib/perl5/site_perl .).
       > BEGIN failed--compilation aborted.
       > make: *** [Makefile:742: install_sax_pureperl] Error 2
       For full logs, run 'nix log /nix/store/1627z6csaj2c14bkwvlkl762pwlnknmc-perl5.34.0-XML-SAX-aarch64-unknown-linux-gnu-1.02.drv'.
error: 1 dependencies of derivation '/nix/store/88asqca7s7wklii3ixlk2gdqk22an3pw-perl5.34.0-Dist-CheckConflicts-aarch64-unknown-linux-gnu-0.11.drv' failed to build
error: 1 dependencies of derivation '/nix/store/v65awrcqfj6xbdbr0lz9b09y4r3l3lkd-perl5.34.0-File-Copy-Recursive-aarch64-unknown-linux-gnu-0.45.drv' failed to build
error: 1 dependencies of derivation '/nix/store/zpv6swakdzx1yiicysbjy6yw0jf78ilj-perl5.34.0-Module-Implementation-aarch64-unknown-linux-gnu-0.09.drv' failed to build
error: 3 dependencies of derivation '/nix/store/5nia7gs8cbc80j4iy2p04fxrvbbk588p-perl5.34.0-Mouse-aarch64-unknown-linux-gnu-2.5.10.drv' failed to build
error: 1 dependencies of derivation '/nix/store/zjry1hff0p1vwcp3j3lypg9ha62356ba-perl5.34.0-B-Hooks-EndOfScope-aarch64-unknown-linux-gnu-0.24.drv' failed to build
error: 1 dependencies of derivation '/nix/store/77g82l4lnsnc073npvzx2hgrla2nvhrl-perl5.34.0-Module-Runtime-Conflicts-aarch64-unknown-linux-gnu-0.003.drv' failed to build
error: 2 dependencies of derivation '/nix/store/py6z71mgmlwcp0asv85j4b0lpr1vwkzh-perl5.34.0-Package-Stash-aarch64-unknown-linux-gnu-0.39.drv' failed to build
error: 2 dependencies of derivation '/nix/store/z23w7yjkh94mzqxqgmy2qs3vyb7pygkj-perl5.34.0-Class-Load-aarch64-unknown-linux-gnu-0.25.drv' failed to build
error: 1 dependencies of derivation '/nix/store/czr4d59vfbxya7x8ywynq4nz48mmlxja-perl5.34.0-Devel-OverloadInfo-aarch64-unknown-linux-gnu-0.005.drv' failed to build
error: 3 dependencies of derivation '/nix/store/wkcahnlqaxsr4k2i1c7fnbzx3w36ynkf-perl5.34.0-Package-DeprecationManager-aarch64-unknown-linux-gnu-0.17.drv' failed to build
error: 2 dependencies of derivation '/nix/store/m00ycjxlim8d4942x6cgix44zwhp0dn8-perl5.34.0-namespace-clean-aarch64-unknown-linux-gnu-0.27.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qcwbbz2xkb8nzyb9x7bfnzrjrxl20apk-perl5.34.0-Class-Load-XS-aarch64-unknown-linux-gnu-0.10.drv' failed to build
error: builder for '/nix/store/pz5y7idw4n88c82m3qklgd61ijz4pqw3-perl5.34.0-XS-Object-Magic-aarch64-unknown-linux-gnu-0.05.drv' failed with exit code 255;
       last 10 log lines:
       > Compilation failed in require at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Handle.pm line 268.
       > BEGIN failed--compilation aborted at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Handle.pm line 268.
       > Compilation failed in require at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Seekable.pm line 100.
       > BEGIN failed--compilation aborted at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/Seekable.pm line 100.
       > Compilation failed in require at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/File.pm line 132.
       > BEGIN failed--compilation aborted at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/IO/File.pm line 132.
       > Compilation failed in require at /nix/store/ki5cfix0faimxska2f211lkfxmf1r62g-perl5.34.0-ExtUtils-Depends-aarch64-unknown-linux-gnu-0.8000/lib/perl5/site_perl/5.34.0/ExtUtils/Depends.pm line 114.
       > BEGIN failed--compilation aborted at /nix/store/ki5cfix0faimxska2f211lkfxmf1r62g-perl5.34.0-ExtUtils-Depends-aarch64-unknown-linux-gnu-0.8000/lib/perl5/site_perl/5.34.0/ExtUtils/Depends.pm line 114.
       > Compilation failed in require at Makefile.PL line 11.
       > BEGIN failed--compilation aborted at Makefile.PL line 11.
       For full logs, run 'nix log /nix/store/pz5y7idw4n88c82m3qklgd61ijz4pqw3-perl5.34.0-XS-Object-Magic-aarch64-unknown-linux-gnu-0.05.drv'.
error: builder for '/nix/store/zl86h0pd297v7gzj5cx8pfja54lzg8mx-perl5.34.0-AnyEvent-I3-aarch64-unknown-linux-gnu-0.17.drv' failed with exit code 255;
       last 10 log lines:
       > patching ./t/01-workspaces.t...
       > Can't load module Fcntl, dynamic loading not available in this perl.
       >   (You may need to build a new perl executable which either supports
       >   dynamic loading or has the Fcntl module statically linked into it.)
       >  at inc/Module/Install/Makefile.pm line 6.
       > Compilation failed in require at inc/Module/Install/Makefile.pm line 6.
       > BEGIN failed--compilation aborted at inc/Module/Install/Makefile.pm line 6.
       > Compilation failed in require at inc/Module/Install.pm line 309.
       > The 'makemaker_args' method does not exist in the 'inc' path!
       > Please remove the 'inc' directory and run Makefile.PL again to load it.
       For full logs, run 'nix log /nix/store/zl86h0pd297v7gzj5cx8pfja54lzg8mx-perl5.34.0-AnyEvent-I3-aarch64-unknown-linux-gnu-0.17.drv'.
error: 1 dependencies of derivation '/nix/store/sk72ipir1kagd14wa8hd9lm0n5y675c4-perl5.34.0-Moo-aarch64-unknown-linux-gnu-2.004004.drv' failed to build
error: 5 dependencies of derivation '/nix/store/88a9ay7xg7xbhhngf6fzn752ndsji5yv-perl5.34.0-Test-CleanNamespaces-aarch64-unknown-linux-gnu-0.24.drv' failed to build
error: 7 dependencies of derivation '/nix/store/dwdchivl62xqgj8qsra0ddvxnlclbb63-perl5.34.0-Moose-aarch64-unknown-linux-gnu-2.2013.drv' failed to build
error: 2 dependencies of derivation '/nix/store/53bysw158ydp3fgjx2n8si7r4x5hsm84-perl5.34.0-Any-Moose-aarch64-unknown-linux-gnu-0.27.drv' failed to build
error: 2 dependencies of derivation '/nix/store/sp00gymabw2biy5zvb419azr1pyrjwj0-perl5.34.0-MouseX-NativeTraits-aarch64-unknown-linux-gnu-1.09.drv' failed to build
error: 1 dependencies of derivation '/nix/store/7j0z4bcdqa53n75ppmqrzbh878dgvz0v-perl5.34.0-Pegex-aarch64-unknown-linux-gnu-0.75.drv' failed to build
error: 3 dependencies of derivation '/nix/store/65vs7p7wjiwr4dxa1rwx5d4zxcbm2g3z-perl5.34.0-Inline-C-aarch64-unknown-linux-gnu-0.81.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rs7dy9057ai6jg1dlm6jpznd6lnh2bhk-perl5.34.0-XML-SAX-Expat-aarch64-unknown-linux-gnu-0.51.drv' failed to build
error: 1 dependencies of derivation '/nix/store/wcns0057d7fahwz406fci4bfpfjarqzj-perl5.34.0-XML-Simple-aarch64-unknown-linux-gnu-2.25.drv' failed to build
error: 1 dependencies of derivation '/nix/store/x143qrszv13f83nqnywchz3f8gg5vwzp-perl5.34.0-XML-Descent-aarch64-unknown-linux-gnu-1.04.drv' failed to build
error: 5 dependencies of derivation '/nix/store/1qjac8cq79mafx1ws0nksj57hbrb5184-perl5.34.0-X11-XCB-aarch64-unknown-linux-gnu-0.18.drv' failed to build
error: builder for '/nix/store/g2fhih397cygschbai8ddk4z180adand-ghostscript-aarch64-unknown-linux-gnu-9.53.3.drv' failed with exit code 1;
       last 10 log lines:
       > checking for x86_64-unknown-linux-gnu-cc... no
       > checking for cc... no
       > checking for x86_64-unknown-linux-gnu-cl.exe... no
       > checking for cl.exe... no
       > checking for x86_64-unknown-linux-gnu-clang... no
       > checking for clang... no
       > configure: error: in `/build/ghostscript-9.53.3/auxtmp1411312588':
       > configure: error: no acceptable C compiler found in $PATH
       > See `config.log' for more details
       > configure: error: Recursive call to configure script failed
       For full logs, run 'nix log /nix/store/g2fhih397cygschbai8ddk4z180adand-ghostscript-aarch64-unknown-linux-gnu-9.53.3.drv'.
error: 1 dependencies of derivation '/nix/store/jmw6b2p8xd2480a29bj5l7v0ir7li93g-fonts.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/kycgswrjjsw5a1q31mxsq1gajgdz6ich-xvfb-run-aarch64-unknown-linux-gnu.drv' failed to build
error: builder for '/nix/store/c8736gckp9vqmg68ii81n2sva4yjrdg5-graphviz-aarch64-unknown-linux-gnu-2.49.3.drv' failed with exit code 2;
       last 10 log lines:
       > make[5]: Leaving directory '/build/source/cmd/dot'
       > make[4]: *** [Makefile:1230: install-exec-am] Error 2
       > make[4]: Leaving directory '/build/source/cmd/dot'
       > make[3]: *** [Makefile:1167: install-am] Error 2
       > make[3]: Leaving directory '/build/source/cmd/dot'
       > make[2]: *** [Makefile:584: install-recursive] Error 1
       > make[2]: Leaving directory '/build/source/cmd'
       > make[1]: *** [Makefile:831: install-recursive] Error 1
       > make[1]: Leaving directory '/build/source'
       > make: *** [Makefile:1140: install] Error 2
       For full logs, run 'nix log /nix/store/c8736gckp9vqmg68ii81n2sva4yjrdg5-graphviz-aarch64-unknown-linux-gnu-2.49.3.drv'.
error: 6 dependencies of derivation '/nix/store/50n1kbqrzl2ilqg816ny8m6rzh4mb3pc-i3-aarch64-unknown-linux-gnu-4.20.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4ygzd2c2x0b30dw3hc850z5rs64g1p4b-itstool-aarch64-unknown-linux-gnu-2.0.6.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1xl5rswxmllyrgs1g1ny1ydyyw0mxxr6-docbook2X-aarch64-unknown-linux-gnu-0.8.8.drv' failed to build
error: build of '/nix/store/1qjac8cq79mafx1ws0nksj57hbrb5184-perl5.34.0-X11-XCB-aarch64-unknown-linux-gnu-0.18.drv', '/nix/store/1xl5rswxmllyrgs1g1ny1ydyyw0mxxr6-docbook2X-aarch64-unknown-linux-gnu-0.8.8.drv', '/n
ix/store/4ygzd2c2x0b30dw3hc850z5rs64g1p4b-itstool-aarch64-unknown-linux-gnu-2.0.6.drv', '/nix/store/50n1kbqrzl2ilqg816ny8m6rzh4mb3pc-i3-aarch64-unknown-linux-gnu-4.20.drv', '/nix/store/c8736gckp9vqmg68ii81n2sva4yj
rdg5-graphviz-aarch64-unknown-linux-gnu-2.49.3.drv', '/nix/store/g2fhih397cygschbai8ddk4z180adand-ghostscript-aarch64-unknown-linux-gnu-9.53.3.drv' failed
```

Artturin committed:
ghostscript: fix cross-compile

zlib has to be in nativeBuildInputs too because its run during the
build, i think.

Artturin committed:
perlPackages.ExtUtilsPkgConfig: fix cross-compiling

Artturin commented:
`pkgsCross.aarch64-multiplatform.perlPackages.ModuleBuild` failure is weird
```
Attempt to reload Scalar/Util.pm aborted.
Compilation failed in require at /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/Data/Dumper.pm line 314.
Couldn't run Build.PL: No such file or directory at lib/Module/Build/Compat.pm line 341.
```

according to https://stackoverflow.com/a/62842326 its trying to load
`/nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/auto/List/Util/Util.so`
```
file /nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/auto/List/Util/Util.so
/nix/store/m02pdpvpfykm66bzs4y9acab37h4kwxi-perl-5.34.0-aarch64-unknown-linux-gnu/lib/perl5/5.34.0/aarch64-linux/auto/List/Util/Util.so: ELF 64-bit LSB shared object, ARM aarch64, version 1 (SYSV), dynamically linked, not stripped
```

Artturin commented:
most of the perl failures are due to https://github.com/NixOS/nixpkgs/pull/142349#issuecomment-948205337
> lack of dynamic loading for our cross-compiled Perl
(idk if there's a issue for that yet)

Artturin commented:
ofborg-eval-lib-tests fails for some reason
```
FAIL: tests/du/basic
====================

--- exp 2021-11-04 18:31:28.992008206 +0000
+++ out 2021-11-04 18:31:28.953008138 +0000
@@ -1,7 +1,7 @@
 1      d/sub/2
 2      d/sub
-9      d/1
-12     d
+1      d/1
+4      d
 ===
 2      d/sub
-10     d
+2      d
FAIL tests/du/basic.sh (exit status: 1)
```
but `nix build ".#coreutils" --rebuild` completes locally

Artturin commented:
@ofborg eval

SuperSandro2000 commented on pkgs/applications/version-management/git-and-tools/git/default.nix:
```diff
@@ -73,7 +73,7 @@ stdenv.mkDerivation {
     ++ lib.optionals withManual [ asciidoc texinfo xmlto docbook2x
          docbook_xsl docbook_xml_dtd_45 libxslt ];
-  buildInputs = [curl openssl zlib expat cpio libiconv]
+  buildInputs = [curl openssl zlib expat cpio libiconv bash ]
```
```suggestion
  buildInputs = [ curl openssl zlib expat cpio libiconv bash ]
```

SuperSandro2000 commented on pkgs/tools/text/sgml/opensp/default.nix:
```diff
@@ -32,10 +32,14 @@ stdenv.mkDerivation rec {
+    docbook_xml_dtd_412
+    docbook_xsl
+  ] ++ lib.optionals stdenv.isCygwin [ autoconf automake libtool ];
 
```
```suggestion
```

SuperSandro2000 commented on pkgs/tools/graphics/graphviz/base.nix:
```diff
@@ -41,11 +41,16 @@ stdenv.mkDerivation {
-    libpng libjpeg expat bison libtool fontconfig gd gts libdevil flex pango
-    gettext
+    libpng libjpeg expat fontconfig gd gts libdevil pango bash
   ] ++ optionals (xorg != null) (with xorg; [ libXrender libXaw libXpm ])
```
```suggestion
  ] ++ (with xorg; [ libXrender libXaw libXpm ])
```

Artturin replied:
graphviz still doesn't cross compile so i'll have to make a separate pr for that so i'll do some refactoring in it

SuperSandro2000 commented on pkgs/servers/pulseaudio/default.nix:
```diff
@@ -40,7 +40,8 @@ stdenv.mkDerivation rec {
 
-  nativeBuildInputs = [ pkg-config autoreconfHook makeWrapper perlPackages.perl perlPackages.XMLParser ];
+  nativeBuildInputs = [ pkg-config autoreconfHook makeWrapper perlPackages.perl perlPackages.XMLParser ]
+      ++ lib.optionals stdenv.isLinux [ glib ];
```
```suggestion
    ++ lib.optionals stdenv.isLinux [ glib ];
```

Artturin replied:
thanks

Artturin commented:
first 3 run fine but the plasma5 one fails because firefox is currently broken on staging
```
nix-instantiate --eval --strict --show-trace ./maintainers/scripts/eval-release.nix
nix build -f ./lib/tests/release.nix
nix-build ./nixos/release-combined.nix -A nixos.iso_minimal.x86_64-linux
nix-build ./nixos/release-combined.nix -A nixos.iso_plasma5.x86_64-linux
```

Artturin commented:
@grahamc i can't repro the ofborg-eval-list-tests failure locally https://gist.github.com/GrahamcOfBorg/daf26c3626990211095411d106e3c9b9

vcunat commented:
On a quick glance it just got interrupted due to being a too large rebuild.

Artturin commented:
the eval test has finally completed, i'll merge in a day if there are no further comments

SuperSandro2000 commented on pkgs/tools/text/groff/default.nix:
```diff
@@ -106,10 +106,8 @@ stdenv.mkDerivation rec {
-  '' + lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform) ''
     find $perl/ -type f -print0 | xargs --null sed -i 's|${buildPackages.perl}|${perl}|'
   '';
-
```
I would like to see this empty line back.

SuperSandro2000 reviewed (commented):
Can we set more strictDeps?

Artturin commented:
> Can we set more strictDeps?

yes but i'd like to do that in a separate pr after this is merged

SuperSandro2000 commented:
I looked through every file and didn't see any obvious code smell. If anything major breaks we will notice it in the staging cycle. Here goes nothing.
